### DISCUSSION TITLE
代码高亮无效

### DISCUSSION THREAD

**Discussion by f88083** [YunYouJun/hexo-theme-starry#383] [Q&A 疑难杂症]
抱歉打扰了，经过两天的搜索与调试，我实在找不出代码无法高亮的原因，恳请大佬们指点迷津。

结果一直是这样：
![image](https://github.com/YunYouJun/hexo-theme-yun/assets/56676355/4c11e64e-4ec2-4768-a6e1-aeca109b531e)

首先我完全按照文档中的方式配置，如下图：

![image](https://github.com/YunYouJun/hexo-theme-yun/assets/56676355/b2510463-7d38-47a6-9414-9ccb3e94b2bb)

也尝试了使用 codeblock，复制的按钮也无法显示

![image](https://github.com/YunYouJun/hexo-theme-yun/assets/56676355/d09b955d-7663-446a-861c-9a6f019f8364)

这里的 base 是我尝试在 head 引入 Prism 的 CSS，也尝试过[文档里推荐](https://cdn.jsdelivr.net/npm/prismjs@1.29.0/themes/prism.css)的主题

![image](https://github.com/YunYouJun/hexo-theme-yun/assets/56676355/d005c29c-ca45-4b11-ac83-1c8b0f5bdcec)

我也阅读过 discussion 的可能解法，尤其是最后一篇，尝试过里面的写法，也用 head 导入过 tomorrow，或是不导入直接使用 tomorrow 都未能成功。或是使用 hexo clean && hexo s 重新构建网站也无法。

![image](https://github.com/YunYouJun/hexo-theme-yun/assets/56676355/a55282ac-23b3-4924-9523-0bacd485f35f)

而我的测试 md 文件内容如下：

![image](https://github.com/YunYouJun/hexo-theme-yun/assets/56676355/c4edadb6-74d6-442b-a1aa-f7951f2c150c)

由衷感谢作者的主题，十分美观实用，如果能有解决办法就太好了，谢谢！

**Comment by YunYouJun** [comment]
Hello，我最近正在搬家，可能暂时没有时间查看调试。

可以考虑尝试 [valaxy](https://github.com/YunYouJun/valaxy) 与 [valaxy-theme-yun](https://github.com/YunYouJun/valaxy/tree/main/packages/valaxy-theme-yun)，我对主题和底层的功能进行了重构，基于 [shiki](https://github.com/shikijs/shiki) 集成提供了多主题及与 VSCode 一致的高亮体验，不再需要第三方插件额外配置。

---

或者我下周将会尝试复现查看该问题。

**Reply by f88083** [reply]
好的非常感謝您的回覆，您方便的時候再查看就好了，感恩。順帶一提，我後來使用了Fluid這個主題發現也使用不了prism，也是無法高亮，換成highlight就可以成功高亮代碼...不知道究竟哪裡出了問題

**Comment by f88083** [comment]
> Hello，我最近正在搬家，可能暂时没有时间查看调试。
> 
> 可以考虑尝试 [valaxy](https://github.com/YunYouJun/valaxy) 与 [valaxy-theme-yun](https://github.com/YunYouJun/valaxy/tree/main/packages/valaxy-theme-yun)，我对主题和底层的功能进行了重构，基于 [shiki](https://github.com/shikijs/shiki) 集成提供了多主题及与 VSCode 一致的高亮体验，不再需要第三方插件额外配置。
> 
> 或者我下周将会尝试复现查看该问题。

@YunYouJun #381  找到原因了，是最近hexo 7.0.0的語法更動所造成 - [連結](https://github.com/hexojs/hexo/issues/5351)
需使用
```yml
syntax_highlighter: prismjs # highlight.js | prismjs | <empty>
```
而非
```yml
highlight:
  enable: false

# 啟用 prism
prismjs:
  enable: true
```
